src/main/java/bio/terra/service/dataset/DatasetSummary.java

```diff
@@ -158,4 +161,31 @@ public DatasetSummary secureMonitoringEnabled(boolean secureMonitoringEnabled) {
     this.secureMonitoringEnabled = secureMonitoringEnabled;
     return this;
   }
+
+  public CloudPlatform getCloudPlatform() {
```

Comment:
You probably want to update the getter for dataset and snapshot to delegate to the summary object for these (for backwards compatibility)

Reply:
I'm confused. This is the summary object.

Reply:
Sorry, so for instance in the dataset object, there's currently a `getProjectResource` method.  It should be rewired to return the value from the summary object (which is a member of the Dataset class) and the setter should be removed (which will require some changes to the Dataset service)

Reply:
Ah yes that makes sense.

Reply:
Sorry, should we be returning the entire `GoogleProjectResource`/`AzureDeploymentResource` in the summary, or just the project name/storage account name?

Reply:
I think it's fine to return the full thing (since we can get it all from the single sql request right?). The main thing I want to avoid is needing to make more calls to the DB for enumeration